11_gan.py

```diff
@@ -37,40 +38,74 @@ def xavier_initializer(shape):
 }
 
 def G(z, w=g_weights):
-    h1 = tf.nn.relu(tf.matmul(z, w['w1']) + w['b1'])
-    return tf.sigmoid(tf.matmul(h1, w['out']) + w['b2'])
+    # here tanh is better than relu
+    h1 = tf.tanh(tf.matmul(z, w['w1']) + w['b1'])
+    # pixel output is in range [0, 255]
+    return tf.sigmoid(tf.matmul(h1, w['out']) + w['b2']) * 255
 
 def D(x, w=d_weights):
-    h1 = tf.nn.relu(tf.matmul(x, w['w1']) + w['b1'])
-    return tf.sigmoid(tf.matmul(h1, w['out']) + w['b2'])
+    # here tanh is better than relu
+    h1 = tf.tanh(tf.matmul(x, w['w1']) + w['b1'])
+    h2 = tf.matmul(h1, w['out']) + w['b2']
+    return h2 # use h2 to calculate logits loss
 
 def generate_z(n=1):
     return np.random.normal(size=(n, z_size))
 
 sample = G(z)
 
 # Objective functions
-G_objective = -tf.reduce_mean(tf.log(D(G(z))))
-D_objective = -tf.reduce_mean(tf.log(D(X)) + tf.log(1 - D(G(z))))
+def sigmoid_cross_entropy_with_logits(x, y):
```

Comment:
Why did you add the wrapper? Why not just using sigmoid_cross_entropy_with_logits?